src/Savegame/Country.cpp

```diff
@@ -333,4 +368,76 @@ bool Country::canBeInfiltrated()
 	return false;
 }
 
+namespace
+{
+
+int checkBoundsAndGet(int index, const std::vector<int>& element)
+{
+	if (index < 0 || index > element.size()) {
+		Log(LOG_WARNING) << "Attempted to get invalid data point.";
```

Comment:
Do not spam logs when error happens

Reply:
? Getting a value out of range is likely never desired behavior. You would just rather it fail silently? `getMonthsTracked` is exposed so the scripting context can know the size of the tracked data.

Reply:
if mod is bugged and call multiple times this function then we could flood logs (maybe in this case it would be hard to archive this but this should be rule how we handle messages in scripts).

Reply:
I mean, if a script is doing something wrong the user/developer really should be informed about that, *especially* if it is wrong in such a way that it ends up flooding a bunch of error messages (this probably means something is *really* wrong). So I guess I just disagree fundamentally here. 

Frankly, I think there is an argument to be made that we should just throw an exception and *die* in this place if a scripting error occurs so that the script writer can definitely be alerted to it. But since there is a reasonable way to recover from this sort of error, I thought recovering and alerting was a better approach.

The only reason to error silently is if the user making an erroneous call was desirable behavior. Say we wanted to let the user easily query all 12 months of data regardless of whether there is data. I could see an argument for this, but then the problem becomes how do we distinguish between a return result that is potentially valid data (like 0) and a return result that indicates an absence of data (currently returning -1, since under base logic none of these values should ever be negative). But if a script's logic is expanded to handle this distinction, it might as well instead expand not to make erroneous calls in the first place.

Reply:
I get you in general, but this specific case is problematic. Take the example I used earlier. I want to base my logic on the last three months of alien activity, but what if there is only one month of data? If I silently return 0 for the absent months without telling the user that I don't have data for these months, they could average two 0's with the one month of data I have and get an incorrect result. But at the same time, zero is also possibly a valid (and common) result for UFO/XCom activity in a region, so it is not possible to distinguish between "no data" and "we have data, and the value is zero" by returning zero in these cases.

That is to say, "we return the data if we have it, and zero if we don't" is a bad contract if data can potentially also contain zero values, which this data can. Returning an out-of-bounds value (-1) is a slightly better contract. But still, since doing so probably indicates the user is doing something in error, I think it's best to warn them (with an error message) is probably the best behavior.

Reply:
If want have this calculate his way then you need add `if` to handle start of game.
I personally dislike calculate average in this way as weights for each data will change from 1 to 1/2, then to 1/3 etc. when months progress.
This mean for starting months average will jump more that after year.
This will could cause problems if you get bad roll in on start of game, if `100` average is "bad" consider sitation that have two events that "cost" `50` each. If both roll in first month (day day 1 and 31) you get `100` average score , if one move one day ahead, then first month will have average `50` and second month will have average `50` too.
In my way of calculating, EVERY month will have close to `33` average, and day when score is calculated will not affect heavy average (first month could have bit less and this is good) .

if we return `-1` lot of naive way to count average will fail and even bugs will be more visible, with `0` some "bugs" will "work" correctly or nor very noticeable.

Reply:
I guess we disagree about what is the best default behavior here. But to me, the situation is the user has asked us a bad question. There really isn't any reasonable contract we can form for correct behavior in the face of the question, "Give me data for a month you do not have," since correct behavior depends on how the user wants to handle the absence of data.

So to me, the only appropriate response is to inform the user that they asked us an incorrect question via some sort of error message/return value. I mean, think about how almost all programming languages handle the situation. Almost universally, they will return an error state if you ask for an out-of-range index. I think following this pattern and being upfront and vocal about the limitations of this method is the best way to proceed.

But, I don't want to bikeshed such a trivial issue. Is this a pattern followed elsewhere in the scripting? I can't think of any instances, but my knowledge is hardly encyclopedic.